addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -507,21 +515,25 @@ def _format_route_info(self, rules, rules_delay, warehouse, product, bom, quanti
         return {}
 
     @api.model
-    def _get_availabilities(self, product, quantity, product_info, bom_key, quantities_info, level, ignore_stock=False, components=False):
+    def _get_availabilities(self, product, quantity, product_info, bom, quantities_info, level, ignore_stock=False, components=False, report_line=False):
         # Get availabilities according to stock (today & forecasted).
         stock_state, stock_delay = ('unavailable', False)
         if not ignore_stock:
             stock_state, stock_delay = self._get_stock_availability(product, quantity, product_info, quantities_info)
 
         # Get availabilities from applied resupply rules
         components = components or []
-        route_info = product_info[product.id].get(bom_key)
+        route_info = product_info[product.id].get(bom.id)
         resupply_state, resupply_delay = ('unavailable', False)
         if product.detailed_type != 'product':
             resupply_state, resupply_delay = ('available', 0)
         elif route_info:
             resupply_state, resupply_delay = self._get_resupply_availability(route_info, components)
 
+        if resupply_state == "unavailable" and route_info == {} and components and report_line:
```

Comment:
I read in the task that this was meant for kits availabilities to be displayed from their components even without having a route defined. This line affects also regular bom, is this the behavior we want?

Reply:
Yes, Mathieu specified it orally but did not write it down on the task. I will modify it now.